lib/elixir/lib/kernel.ex

```diff
@@ -2703,7 +2703,15 @@ defmodule Kernel do
   """
   defmacro left |> right do
     [{h, _}|t] = Macro.unpipe({:|>, [], [left, right]})
-    :lists.foldl fn {x, pos}, acc -> Macro.pipe(acc, x, pos) end, h, t
+    :lists.foldl fn {x, pos}, acc ->
+      # TODO: raise an error in `Macro.pipe` when we drop unary operator support in pipes
+      case Macro.pipe_warning(x) do
```

Comment:
It doesn't matter much right now. It is a temporary function, so we are fine either way. :)